### PR TITLE
Add SeaTunnel Engine 

### PR DESCRIPTION
**Pull request by EricJoy2048**
## Purpose of this pull request

Added a new module named `seatunnel-engine`. 

Same simple code about  https://github.com/apache/incubator-seatunnel/issues/1954

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by xleoken** [comment]
hi @gaojun2048, looked over your patch, it seems like a scaled-down flink, can the engine run on other resource framework? like yarn or kerberos?

**Comment by EricJoy2048** [comment]
> hi @gaojun2048, looked over your patch, it seems like a scaled-down flink, can the engine run on other resource framework? like yarn or kerberos?

Yes, it looks like a scaled down Flink. Flink is very mature in task scheduling and running. In order to quickly complete the functions of a basic version, I tried to refer to Flink in my design, because many designs in Flink are easier to be accepted. When we can run basic task scheduling and task execution, Next, we should focus on the problems that seatunnel engine really needs to solve. The design goal of seatunnel engine is to solve the problems in these data synchronization scenarios, rather than developing a computing engine like Flink.

> can the engine run on other resource framework? like yarn or kerberos?

Yes, I hope the seatunnel engine can support yarn and k8s, but I can't do such a huge job alone. I need the support and help of the community.

File: seatunnel-engine/src/test/java/org/apache/seatunnel/engine/maintest/SeaTunnelEngineMainTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    protected final Logger logger = LoggerFactory.getLogger(SeaTunnelEngineMainTest.class);
+
+    @Before
+    public void before() throws Exception {
+    }
+
+    @After
+    public void after() throws Exception {
+    }
+
```
Comment (CalvinKirs) [inline seatunnel-engine/src/test/java/org/apache/seatunnel/engine/maintest/SeaTunnelEngineMainTest.java:50]: If it doesn't make sense to me.

File: seatunnel-engine/src/test/java/org/apache/seatunnel/engine/execution/TaskExecutionTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        TaskExecution taskExecution = new TaskExecution(executorService, null);
+
+        taskExecution.submit(taskInfo);
+
+        Thread.sleep(500L);
+
+        Assert.assertEquals(taskExecution.aliveTaskSize(), 1);
+
+        Thread.sleep(5000L);
+
```
Comment (CalvinKirs) [inline seatunnel-engine/src/test/java/org/apache/seatunnel/engine/execution/TaskExecutionTest.java:80]: Is this test stable?

Comment (EricJoy2048) [reply seatunnel-engine/src/test/java/org/apache/seatunnel/engine/execution/TaskExecutionTest.java:80]: yes.

**Pull request closed without merging**